client/block/store.go

```diff
@@ -0,0 +1,155 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+package block
+
+import (
+	"fmt"
+	"path/filepath"
+
+	dbm "github.com/cometbft/cometbft-db"
+	tmstore "github.com/cometbft/cometbft/proto/tendermint/store"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	"github.com/cometbft/cometbft/types"
+	"github.com/cosmos/gogoproto/proto"
+)
+
+var storeKey = []byte("blockStore")
+
+// store is the block store struct
+type store struct {
+	dbm.DB
+}
+
+// newStore opens the 'blockstore' db
+// and returns it.
+func newStore(rootDir string, backendType dbm.BackendType) (*store, error) {
+	dataDir := filepath.Join(rootDir, "data")
+	db, err := dbm.NewDB("blockstore", backendType, dataDir)
+	if err != nil {
+		return nil, err
+	}
+
+	return &store{db}, nil
+}
+
+// state returns the BlockStoreState as loaded from disk.
+// If no BlockStoreState was previously persisted, it returns nil.
+func (s *store) state() *tmstore.BlockStoreState {
+	bytes, err := s.Get(storeKey)
+	if err != nil {
+		panic(err)
```

Comment:
are we panacking the node through a cli request ?

Reply:
actually is a cmd to run without the node running. Like the `evmos init`. The panic will not stop the node
But will return error and handle it anyways, so is cleaner